test/src/modules/smoke/shadow/__tests__/shadow.spec.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+describe('shadow DOM', () => {
+    // Smoke test to check that we're running in the right shadow DOM mode
+    if (global['lwc-jest'].nativeShadow) {
+        it('should be using native shadow DOM', () => {
+            // sniff for JSOM's ShadowRoot implementation
+            expect(ShadowRoot.prototype.constructor.toString()).toContain('class ShadowRoot extends globalObject.DocumentFragment');
```

Comment:
If we need to rely on internal details, might as well only rely on our own internal details?
```suggestion
            expect(ShadowRoot.prototype.constructor.toString()).not.toContain('function SyntheticShadowRoot');
```